### ISSUE TITLE
Errors during fake offline Full Threshold

### ISSUE THREAD

**Issue by ChaitanyaKr** [KULeuven-COSIC/SCALE-MAMBA#8]
Hi,
I tried running Full Threshold secret sharing, for 6 parties (on 4 VMs, 2-1-2-1, each with 16GB memory and 4 cores), with fake offline and fake sacrifice phase, but couldn't get the tests to run successfully, specifically the `Programs/tutorial` one.

- Fake offline: Y; Fake sacrifice: N--> crashes just before sacrifice phase
- Fake offline: Y; Fake sacrifice: Y--> crashes after running all assertion tests specified in the .mpc file, giving the `mac_fail` error

Is there some configuration that I'm missing out while running this? I understand that the fake offline phase is only for testing and I'm trying to do this to identify my setups resource bottlenecks.

**Comment by NigelSmart** [comment]
Hi

No idea. We never test with fake offline any more, it was
only there for development purposes. We perhaps should remove
it I think.

Could be memory though. Have you looked at top, or /usr/bin/time -v?

Are you running on one machine? Probably a bad idea. The system
is configured for running on multiple machines (its the "M"
in MPC  :-)   )

Nigel

**Comment by ChaitanyaKr** [comment]
Yes, like I mentioned the 6 parties are running on 4 VMs, all essentially separate machines. The VM with 2 parties causes all of them to eventually crash. 

It is memory for sure, as `htop` tells me. I just wanted to isolate which phase was causing it, if possible.
In case of memory being insufficient, I've seen one player reporting `bad_alloc` and the others to report `send/receive` errors-- which makes sense. I'm not clear as to why the fake phases cause a crash too.

**Comment by NigelSmart** [comment]
Basically, it is probably the ZKPoKs which we do not turn
off when using fake offline.

After all
  a) fake offline was only created for our development
  b) You will need the ZKPoKs when running in a proper system

Suggest switching to six machines

Nigel
